app/src/main/java/org/mozilla/tv/firefox/fxa/FxaRepo.kt

```diff
@@ -225,49 +226,53 @@ fun Profile.toDomainObject(): FxaProfile {
     return FxaProfile(validatedAvatar, validatedDisplayName)
 }
 
-private fun Observable<ADMIntegration.ReceivedTabs>.mapToReceivedTabs(): Observable<ReceivedTabs> = map { admTabs ->
-    ReceivedTabs(
-        urls = admTabs.tabData
-            .map { it.url }
-            .filter { it.isNotBlank() }, // blank URLs may not be possible but we do this just to be safe.
-        sendingDevice = admTabs.device?.let {
-            ReceivedTabs.DeviceMetadata(
-                displayName = it.displayName,
-                deviceType = it.deviceType
-            )
-        }
-    )
-}
-
-private fun Observable<ReceivedTabs>.filterInvalidTabs(sentryIntegration: SentryIntegration): Observable<ReceivedTabs> = filter {
-    return@filter if (it.urls.isNotEmpty()) {
-        true
-    } else {
-        sentryIntegration.captureAndLogError(logger, ReceiveTabException("Received tab event with only blank URLs"))
-        false
-    }
-}
-
 /**
  * A data container for tabs received from a single device via FxA. While we currently support only
  * one tab, this container can actually contain multiple received tabs.
  *
  * This container exists to avoid exposing the underlying [DeviceEvent.TabReceived] events to consumers.
  */
-data class ReceivedTabs constructor(
-    /** The urls sent to this device. */
-    val urls: List<String>,
-
-    /** Metadata about the device that sent the tab, or null if it's unavailable. */
-    val sendingDevice: DeviceMetadata?
+data class FxaReceivedTab(
+    val url: String,
+    val sourceDescription: UnresolvedString,
```

Comment:
nit: I don't really know what `sourceDescription` is. Maybe, `tabReceivedByDeviceExplanation`? This member could also use a kdoc if it's hard to name. Note there's a local variable that shadows this name that we'd probably want to update too.